eyelid-match-ops/src/primitives/poly.rs

```diff
@@ -326,3 +330,76 @@ pub fn poly_split_half(a: &Poly, chunk: usize) -> (Poly, Poly) {
 
     (remainder, quotient)
 }
+
+/// Returns the inverse in the cyclotomic ring, if it exists.
+/// Otherwise, returns an error.
+pub fn inverse(a: &Poly) -> Result<Poly, String> {
+    let mut mod_pol = Poly::zero();
+
+    // TODO: don't recompute the modulus here
+    mod_pol[MAX_POLY_DEGREE] = Coeff::one();
```

Comment:
There’s a function for this now, I’ll make these changes today as part of updating this code for PR #40.

Reply:
Done!